### ISSUE TITLE
Why the initial labelled sample was repeated 20 times in the code?

### ISSUE THREAD

**Issue by xuhaodong1992** [mrottmann/DeepBASS#1]
Hi, great work!!! But when I study the code, I found the initial labelled sample have been trained 20 times as follow (weight_gt was set as 20). So the training set is 20 times larger when the initial model is trained rather than a tiny part . Can you explain the reason?
(weight_gt=20)
for i in range(Niter+1):
    for j in range(1,weight_gt):
      I = np.concatenate((I,I_start),axis=0)

**Comment by mrottmann** [comment]
Sorry, I didn't receive a notification. I'll adjust the settings. Probably this is not important for you anymore, but I'll briefly answer your question.

This up-sampling approach ensures that data with ground truth provided by an oracle is visited more often during training than unlabeled data. If you don't put some emphasis on the ground truth labeled data, it might happen that this important information gets lost during the numerous training epochs. You can find further details in https://arxiv.org/pdf/1803.01216.pdf

Cheers
